### PR TITLE
ref(getter): flatten the getter package tree

### PR DESCRIPTION
**Pull request by technosophos**
This flattens the getter package tree, adds tests, and changes a little
bit of the terminology to follow Go idioms. This also makes much of the
getter API private to begin with. This will give us more flexibility in
the future.

### CODE REVIEW

File: pkg/getter/httpgetter.go [hunk 1]
```diff
@@ -49,9 +47,9 @@ func (g *HTTPGetter) Get(href string) (*bytes.Buffer, error) {
 	return buf, err
 }
 
-//New constructs a valid http/https client as Getter
-func New(URL, CertFile, KeyFile, CAFile string) (getter.Getter, error) {
-	var client HTTPGetter
+// newHttpGetter constructs a valid http/https client as Getter
```
Comment (adamreese) [inline pkg/getter/httpgetter.go]: `s/newHttpGetter/newHTTPGetter/`

File: pkg/getter/httpgetter_test.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+limitations under the License.
+*/
+
+package getter
+
+import (
+	"testing"
+)
+
+func TestHttpGetter(t *testing.T) {
```
Comment (adamreese) [inline pkg/getter/httpgetter_test.go]: what are you testing here?

**Comment by technosophos** [comment]
Along the way of improving tests, found an error in urlutil.go, fixed, and added tests.

File: pkg/getter/plugingetter.go [hunk 3]
```diff
@@ -22,14 +22,37 @@ import (
...
 
-	"k8s.io/helm/pkg/getter"
 	"k8s.io/helm/pkg/helm/environment"
 	"k8s.io/helm/pkg/plugin"
 )
 
-// PluginGetter is a generic type to invoke custom downloaders,
+// collectPlugins scans for getter plugins.
+// This will load plugins according to the environment.
+func collectPlugins(settings environment.EnvSettings) (Providers, error) {
```
Comment (adamreese) [inline pkg/getter/plugingetter.go]: collect them all!

**Review by adamreese — APPROVED** [review]

**Pull request merged**